icicle/CMakeLists.txt

```diff
@@ -7,7 +7,9 @@ set(CMAKE_CUDA_STANDARD_REQUIRED TRUE)
 set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
 # add the target cuda architectures
 # each additional architecture increases the compilation time and output file size
-if (NOT DEFINED CMAKE_CUDA_ARCHITECTURES)
+if (${CMAKE_VERSION} VERSION_LESS "3.24.0")
+    set(CMAKE_CUDA_ARCHITECTURES ${CUDA_ARCH})
```

Comment:
Yes, it turns out that `CUDA_ARCH` is undefined. When line 11 runs, however, `CMAKE_CUDA_ARCHITECTURES` retains its original value.  As such, perhaps a better way is:

```
if (${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.24.0")
    set(CMAKE_CUDA_ARCHITECTURES native)
endif ()
```

Reply:
We should have a flexible way to provide/override the compiler's chosen arch. Maybe the following would be best for that:
```
if(DEFINED CUDA_ARCH) // user defined arch takes priority
    set(CMAKE_CUDA_ARCHITECTURES ${CUDA_ARCH})
elseif(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.24.0") // otherwise, use native to detect GPU arch
    set(CMAKE_CUDA_ARCHITECTURES native)
endif() // otherwise fallback to compiler's chosen arch
```